### PR TITLE
Migration of the source term and analytical solution parameter sections

### PR DESCRIPTION
**Pull request by CaroleAnneDaunais**
# Description of the problem

The source term and analytical solution sections were still to be merged. 

# Description of the solution

I added both files in this PR.

# Future changes

As I was checking the defaults parameters for the source term section, I noticed the default value for enable is true which should be changed to false. I also noticed it's definition is to "Enable calculation for analytical solution and L2 error" which should also be fixed.

# Comments

I added the post_processing name to the cfd.rst file since I hadn't done it previously.

### CODE REVIEW

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      set Function expression = 0
+    end
+    subsection phase
+      set Function expression = 0
+    end
+   end
+
+* The ``enable`` parameter sets if the problem has an analytical solution and enables the calculation of the analytical solution and the norm of the L2 error.
+
+* The ``verbosity`` parameter enables printing the L2 error after each mesh refinement if set to ``verbose``, if ``enable`` is ``true``.
```
Comment (blaisb) [inline doc/source/parameters/cfd/analytical_solution.rst:31]:
```suggestion
* The ``verbosity`` parameter enables printing the L2 error after each mesh refinement if it is set to ``verbose`` and if ``enable`` is ``true``.
```

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+* The ``filename`` parameter sets the file name to output the L2 error norm if ``enable`` is ``true``.
+
+If there is an analytical solution for velocity and pressure, enter the ``uvwp`` subsection.
+
+* The ``Function expression`` parameter sets the expression for the analytical solution in regards to *u*, *v* and *p* for a 2D simulation and to *u*, *v*, *w* and *p* for a 3D simulation.
```
Comment (blaisb) [inline doc/source/parameters/cfd/analytical_solution.rst:37]:
```suggestion
* The ``Function expression`` parameter sets the expression for the analytical solution in regards to :math:`u`, :math:`v` and :math:`p` for a 2D simulation and to :math:`u`, :math:`v`, :math:`w` and :math:`p` for a 3D simulation.
```

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+If there is an analytical solution for the fluid's phase, enter the ``phase`` subsection.
+
+* The ``Function expression`` parameter sets the expression of the phase.
+
+.. note:: 
+    The variables *x*, *y*, *z* (3D) and *t* (time-dependant) can be used in the function expressions.
+
+In all four last subsections, you can add a ``Function constants`` parameter that will act as a constant in the ``Function expression``.
+
+You can add a ``Function constants`` parameter that will act as a constant in the ``Function expression``.
```
Comment (blaisb) [inline doc/source/parameters/cfd/analytical_solution.rst:56]:
```suggestion
```

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+* The ``Function expression`` parameter sets the expression of the phase.
+
+.. note:: 
+    The variables *x*, *y*, *z* (3D) and *t* (time-dependant) can be used in the function expressions.
+
+In all four last subsections, you can add a ``Function constants`` parameter that will act as a constant in the ``Function expression``.
+
+You can add a ``Function constants`` parameter that will act as a constant in the ``Function expression``. 
+
+Ex.
```
Comment (blaisb) [inline doc/source/parameters/cfd/analytical_solution.rst:58]:
```suggestion
Ex:
```

File: doc/source/parameters/cfd/source_term.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    subsection tracer
+      set Function expression = 0
+    end
+   end
+
+* The ``enable`` parameter sets if the problem has a source term and enables it's calculation.
+
+If there is an Navier-Stokes source term, enter the ``xyz`` subsection.
+
+* The ``Function expression`` parameters sets the expression for the source term in regards to *x*, *y* and *p* for a 2D simulation and to *x*, *y*, *z* and *p* for a 3D simulation.
```
Comment (blaisb) [inline doc/source/parameters/cfd/source_term.rst:27]:
```suggestion
* The ``Function expression`` parameters sets the expression for the source term. In 2D, the first two terms are the source term for  the *x*, *y* component of the momentum equation. The third term is the mass source term. In 3D, the first three terms are for the *x*, *y* and *z* component of the momentum equation and the fourth term is for the mass source term.
```

**Review by blaisb — COMMENTED** [review]
Some minor comments, but good job!

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    end
+    subsection tracer
+      set Function expression = 0
+    end
+    subsection phase
+      set Function expression = 0
+    end
+   end
+
+* The ``enable`` parameter sets if the problem has an analytical solution and enables the calculation of the analytical solution and the :math:`L^2` norm of the error.
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/analytical_solution.rst:29]:
```suggestion
* The ``enable`` parameter is set to true if the problem has an analytical solution. This enables the calculation of the analytical solution and the :math:`L^2` norm of the error.
```

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+   subsection analytical solution
+    set enable                = true
+    set verbosity             = true
+    set filename              = L2Error
+    subsection phase
+      set Function expression = if(sin(x) > pi, 1, 0)
+    end
+   end
+
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/analytical_solution.rst:97]: Wouldn't it be better to specify that the first condition after the if is applied when it is true and the second one is the else condition?

File: doc/source/parameters/cfd/source_term.rst [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    end
+    subsection heat transfer
+      set Function expression = 0
+    end
+    subsection tracer
+      set Function expression = 0
+    end
+   end
+
+* The ``enable`` parameter sets if the problem has a source term and enables it's calculation.
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/source_term.rst:23]:
```suggestion
* The ``enable`` parameter is set to true if the problem has a source term and enables it's calculation.
```

File: doc/source/parameters/cfd/source_term.rst [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+If there is an Navier-Stokes source term, enter the ``xyz`` subsection.
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/source_term.rst:25]:
```suggestion
If there is a Navier-Stokes source term, enter the ``xyz`` subsection.
```

File: doc/source/parameters/cfd/source_term.rst [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+* The ``Function expression`` parameters sets the expression for the source term. In 2D, the first two terms are the source term for  the *x*, *y* component of the momentum equation. The third term is the mass source term. In 3D, the first three terms are for the *x*, *y* and *z* component of the momentum equation and the fourth term is for the mass source term.
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/source_term.rst:27]:
```suggestion
* The ``Function expression`` parameters sets the expression for the source term. In 2D, the first two terms are the source terms for  the *x*, *y* component of the momentum equation. The third term is the mass source term. In 3D, the first three terms are for the *x*, *y* and *z* component of the momentum equation and the fourth term is for the mass source term.
```

File: doc/source/parameters/cfd/source_term.rst [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+Ex.
+
+.. code-block:: text
+
+   subsection source term
+    set enable                = true
+    subsection heat transfer
+      set Function expression = if(sin(x) > pi, 1, 0)
+    end
+   end
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/source_term.rst:68]: Same comment as before

File: doc/source/parameters/cfd/source_term.rst [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+.. code-block:: text
+
+   subsection source term
+    set enable                = true
+    subsection xyz
+      set Function constants = A=2.0
+      set Function expression = A*y; -A*x; 0
+    end
+   end
+
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/source_term.rst:53]: Same comment as before

File: doc/source/parameters/cfd/analytical_solution.rst [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+   subsection analytical solution
+    set enable                = true
+    set verbosity             = true
+    set filename              = L2Error
+    subsection uvwp
+      set Function constants = A=2.0
+      set Function expression = A*y; -A*x; 0
+    end
+   end
+
```
Comment (tonielgeitani) [inline doc/source/parameters/cfd/analytical_solution.rst:79]: It would be good to specify how many constants can we define per subsection.

**Review by tonielgeitani — APPROVED** [review]

File: doc/source/parameters/cfd/source_term.rst (hunk 12 above)
Comment (CaroleAnneDaunais) [reply doc/source/parameters/cfd/source_term.rst:53]: We should definitely add it, though I have no idea how many.

Comment (blaisb) [reply doc/source/parameters/cfd/source_term.rst:53]: I don't think there is a limit as far as I know...

**Review by blaisb — APPROVED** [review]

**Comment by blaisb** [comment]
@CaroleAnneDaunais Be careful to batch commit the changes when people make suggestions, otherwise it clogs-up the CI.

**Pull request merged**